### PR TITLE
Blair start game transition

### PR DESCRIPTION
**Pull request by BlaiiZong**
implemented error checking and transition for starting the game, does not link to night 0 page

### CODE REVIEW

File: client/src/components/ChooseRole.js [hunk 1]
```diff
@@ -1,61 +1,70 @@
...
+    roles,
+    setRoles,
+    numMafia,
+    setNumMafia,
+    numTownspeople,
+    setNumTownspeople,
+  } = useContext(GameContext);
 
-  const addRole = (index, change) => {
+  const getRoleIndex = (roleId) => {
```
Comment (munjoonteo) [inline client/src/components/ChooseRole.js:19]: You can use a combination of `map` and `indexof` to simplify this

File: client/src/App.js [hunk 2]
```diff
@@ -10,23 +11,25 @@ import { GameContext } from './GameContext';
...
-    <div className="App"> 
+    <div className="App">
       {/* if (page === 'turnSelect' ? <TurnSelectScreen title="Select which player you are!" : <ActionMenu title="Select another player" ) */}
 
       {/* <TurnSelectScreen /> */}
       {/* <ActionMenu /> */}
-      {currentPage == "actionMenu" && <ActionMenu/>}
       {currentPage === 'main' && <MainMenu />}
+      {
+        currentPage === 'displayPlayerRoles' && <DisplayPlayerRoles />
```
Comment (munjoonteo) [inline client/src/App.js]: Well you could just console.log the players right. No need to make a whole page for it.

File: client/src/GameContext.js [hunk 3]
```diff
@@ -37,55 +66,55 @@ const GameContextProvider = ({ children }) => {
     {
       name: 'James',
       alive: true,
-      role: 'barista',
+      role: 'none',
```
Comment (munjoonteo) [inline client/src/GameContext.js:69]: Who will be making this not hardcoded again?

**Review by munjoonteo — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 2 above)
Comment (BlaiiZong) [reply client/src/App.js]: oh ok true; and delete when PR'ing?

File: client/src/GameContext.js (hunk 3 above)
Comment (BlaiiZong) [reply client/src/GameContext.js:69]: I believe Ahnaf when he implements the add player page; I've implemented assigning roles so i "un-hardcoded" that

File: client/src/components/ChooseRole.js (hunk 1 above)
Comment (BlaiiZong) [reply client/src/components/ChooseRole.js:19]: fixed

File: client/src/components/ChooseRole.js [hunk 4]
```diff
@@ -1,61 +1,66 @@
...
+      setRoleError(roleId, false);
     }
     setRoles(newRoles);
   };
 
-  const [errorMsgs, setErrorMsgs] = useState(
-    [...Array(roles.length)].fill(false)
-  );
+  const resetRoleErrors = () => {
+    return roles.reduce(
```
Comment (munjoonteo) [inline client/src/components/ChooseRole.js]: Since I have nothing else to comment I will nitpick on this. This can be an implicit return.

**Review by munjoonteo — CHANGES_REQUESTED** [review]

**Review by munjoonteo — APPROVED** [review]

**Pull request merged**